File: scratchpad/242. Valid Anagram.cpp
#include <iostream>
#include <unordered_map>
#include <string>

class Solution {
public:
    bool isAnagram(std::string s, std::string t) {
        std::unordered_map<char, int> s_map, t_map;

        for (char c : s) {
            s_map[c]++;
        }

        for (char c : t) {
            t_map[c]++;
        }

        return s_map == t_map;
    }
};

int main() {
    Solution solution;
    std::string s = "anagram";
    std::string t = "nagaram";
    
    if (solution.isAnagram(s, t)) {
        std::cout << "The strings are anagrams." << std::endl;
    } else {
        std::cout << "The strings are not anagrams." << std::endl;
    }

    return 0;
}

File: strings/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False

        s_map, t_map = {}, {}

        for char in s:
            if char not in s_map:
                s_map[char] = 1
            else:
                s_map[char] += 1

        for char in t:
            if char not in t_map:
                t_map[char] = 1
            else:
                t_map[char] += 1

        return True if s_map == t_map else False


        #for char in s_map:
        #    if s_map[char] != t_map.get(char, 0):
        #        return False
       
        #return True 